File: gameFramework/src/main/java/at/grueneis/game/framework/FileIO.java
package at.grueneis.game.framework;

import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;

public interface FileIO
{
	public InputStream readAsset(String filename) throws IOException;
	
	public InputStream readFile(String filename) throws IOException;
	
	public OutputStream writeFile(String filename) throws IOException;
}

File: gameFramework/src/main/java/at/grueneis/game/framework/FileIO.kt
package at.grueneis.game.framework;

import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;

public interface FileIO
{
	InputStream readAsset(String filename) throws IOException;
	
	InputStream readFile(String filename) throws IOException;
	
	OutputStream writeFile(String filename) throws IOException;
}
